### ISSUE TITLE
Missing solution?

### ISSUE THREAD

**Issue by arpruss** [midjji/pnp#3]
I am not experienced with p3p and homogeneous coordinates, but I am having some trouble. I feed the following data to p3p:

    cvl::Vector3<double> y1 = { 0,0,1 };
    cvl::Vector3<double> y2 = { 2,0,1 };
    cvl::Vector3<double> y3 = { 0,2,1 };

    cvl::Vector3<double> x1 = { 0,0,0 };
    cvl::Vector3<double> x2 = { 1,0,0 };
    cvl::Vector3<double> x3 = { 0,1,0 };

The "obvious" solution I am expecting would also map x4 = { 1,1,0 } to { 2,2,1 }. And unless I am misunderstanding how this stuff works, which is not unlikely, I think this solution would correspond to:

    T = 0 0 0.5
    R = identity.

But p3p is returning only one solution:

    T = 0 0 0.5625
    R = { { 1, 0, 0.0625 },
         { 0, 1, 0.0625 },
         { -0.0625, -0.0625, 1 } }.

And that maps x4 to { 2.29, 2.29, 1 }, which is not the "obvious" solution.

**Comment by midjji** [comment]
I'm looking into it,

**Comment by arpruss** [comment]
I've also had missing solutions in real life data when using the algorithm for p4p, so it's probably not just a matter of a borderline case.

**Comment by midjji** [comment]
Classic quartic p4p has a lot more problems than this one, but there was a missing case in the innermost solver, I believed it was impossible because it never occurred no matter how many times I ran it, and so carelessly didnt port it. But you have found it :D. I have been looking for this case for ages. 

I have added the missing case, and it nicely enough turns out to slightly improve numerical accuracy at zero performance cost. And does find the expected solution too. So thats nice. I will be porting the fix to the pnp over the weekend.  

We never strictly proved that there could not be additional solutions but I believe it to be the case. Or at least that should be, if the system is implemented correctly, as it will be very soon. The number of solutions for p3p is chaotic however, and not just near degenerate cases, and there are two places left where its possible that they could occur now. However, in practice, like for real data, missing solutions should be exceedingly rare with this method. Its one of the three main advantages compared to other methods. Better numerics and more solutions, all of which should be geometrically valid. At least for numerically non-degenerates. 

I will keep the ticked open until I have ported the fix.
